src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -45,6 +49,10 @@ public class ParameterProvider {
   public static final long MAX_MEMORY_LIMIT_IN_BYTES_DEFAULT = -1L;
   public static final long MAX_CHANNEL_SIZE_IN_BYTES_DEFAULT = 32000000L;
   public static final long MAX_CHUNK_SIZE_IN_BYTES_DEFAULT = 128000000L;
+
+  // Lag related parameters
+  public static final String MAX_CLIENT_LAG_DEFAULT = "1 second";
+  public static final boolean MAX_CLIENT_LAG_ENABLED_DEFAULT = true;
```

Comment:
why do we need this enabled parameter too?

Reply:
Good question: if we find that there's a bug with this new code path we can disable it and go back to the old behavior.

Reply:
Makes sense to me